### ISSUE TITLE
SC2 replay BC [5/6]: policy-agnostic warm-start — fine-tune from BC with ANY policy

### ISSUE THREAD

**Issue by espenhk** [espenhk/gamer-ai#354]
Part of #349. Depends on [2/6] and [4/6]. **This is the "fine-tune using any available policy" requirement.**

## Goal
Make the BC demonstration dataset usable as a warm-start for **every SC2-compatible policy family**, not just the default MLP. The sequence-aware dataset from [2/6] is the policy-agnostic interchange format; each policy family ingests it via the loader view that matches its trainer (flat per-step matrices for memoryless policies, ordered per-episode sequences for recurrent ones).

## Design
Introduce a single opt-in hook (e.g. `BasePolicy.warm_start_from_demonstrations(dataset, obs_spec)` with a no-op default, or a small registry keyed by `policy_type`) so `--bc` / `do_bc` can target whatever `policy_type` is set. Per-family implementations:

| Policy family | Dataset view | Warm-start strategy |
|---|---|---|
| `sc2_reinforce` (**default**), `sc2_neural_net` | flat per-step | Supervised MLP pre-train: CE on fn head + MSE on sigmoid spatial head ([4/6]). |
| `sc2_genetic`, `sc2_cmaes` | flat per-step | Closed-form linear fit into `SC2MultiHeadLinearPolicy`; seed champion / CMA-ES mean. |
| `sc2_neural_dqn` | flat per-step | Pre-fill replay buffer with demonstrations and/or warm-start the Q-net head. |
| `sc2_lstm` | **ordered sequences** | Supervised pre-train of the recurrent net over per-episode sequences with hidden-state carry-over (uses `episode_starts`). |
| `sc2_cnn` | flat per-step (spatial layers) | Supervised pre-train of conv+FC head; requires non-empty `screen_layers`. |
| `epsilon_greedy`, `ucb_q` | flat per-step | Seed tabular Q-table / counts from binned demonstration `(state, action)` visits. |

`fit_bc` from [4/6] gains a `target` switch dispatching to these; each writes the trained weights (+ trainer state) in that policy's own save format so a subsequent `python main.py --game sc2` resumes/fine-tunes normally.

## Acceptance criteria
- For each supported `policy_type`, a BC warm-start produces a weights/trainer file the policy's `load` accepts and the greedy/gradient loop continues from without `--re-initialize`.
- Recurrent (`sc2_lstm`) warm-start consumes ordered sequences and respects episode boundaries.
- Round-trip tests (mocked dataset) per family; clear error if a `policy_type` has no warm-start support.

## Notes
- SB3 policies (`ppo`/`sac`/…) are gated off SC2 — document the exclusion.
- Land order: MLP (`sc2_reinforce`, the default) + linear (`sc2_genetic`) first, then `sc2_lstm` (sequence path), then DQN/CNN/tabular.

## Validation
`python -m pytest tests/test_sc2_replay_bc.py`. Update `tests/README.md`.
